src/rust/engine/graph/src/entry.rs

```diff
@@ -157,9 +202,9 @@ impl<N: Node> Entry<N> {
   pub fn peek(&self) -> Option<Result<N::Item, N::Error>> {
     let state = self.state.lock();
     match *state {
-      EntryState::Completed {
-        ref result, dirty, ..
-      } if !dirty => Some(result.clone()),
+      EntryState::Completed { ref result, .. } if !result.is_dirty() => {
```

Comment:
My understanding was that something like:
```
EntryState::Completed { result: Clean(ref res), .. } => ..
```
would work... but the guard is perfectly fine too.

Reply:
Done! I was having issues working out where the ref needed placing.